mara_pipelines/incremental_processing/file_dependencies.py

```diff
@@ -32,7 +32,7 @@ def update(node_path: [str], dependency_type: str, pipeline_base_path: str, file
         pipeline_base_path: The base directory of the pipeline
         file_dependencies: A list of file names relative to pipeline_base_path
     """
-    with mara_db.postgresql.postgres_cursor_context('mara') as cursor:
+    with mara_db.dbs.cursor_context('mara') as cursor:
```

Comment:
Does that actually mean one can run the mara db on something else than pg? That would be surprising, I remember it was using lot's of PG specific thingies (`VALUES (...)`)?

If not, i would tend to at least mention that this is a PG...

Reply:
My fault, the comment was a bit wrong: The main purpose changing this is to get around the deprecation warning which we implemented in mara-db 4.9.

I tried in the past to use another database system but the main issue is that the node path is saved as a string array. Most dbs do not support arrays. I only now of BigQuery but BigQuery is not something you should use as transactional storage. There are limits and when you reach them, BigQuery will block further transactions...

I actually don't know if it is possible to run a ppipeline on something else than PG. I created #94 as a starting point for people who want to achive this. I see a great potential in this. E.g. automatically create a SQLite db as cache to store historical node costs.

In the README it is written that PostgreSQL is recommended. Is that enough for mentioning that PG should be used?